shaptools/netweaver.py

```diff
@@ -147,6 +151,32 @@ def is_installed(self, sap_instance=None):
         else:
             raise ValueError('provided sap instance type is not valid: {}'.format(sap_instance))
 
+    @classmethod
+    def update_conf_file(cls, conf_file, **kwargs):
+        """
+        Update NW installation config file parameters. Add the parameters if they don't exist
+
+        Args:
+            conf_file (str): Path to the netweaver installation configuration file
+            kwargs (opt): Dictionary with the values to be updated.
+                Use the exact name of the netweaver configuration file
+
+        kwargs can be used in the next two modes:
+            update_conf_file(conf_file, sid='HA1', hostname='hacert01')
+            update_conf_file(conf_file, **{'sid': 'HA1', 'hostname': 'hacert01'})
+        """
+        for key, value in kwargs.items():
+            pattern = '{key}( *)=.*'.format(key=key)
```

Comment:
I think you can change the pattern to:
`pattern = '{key}\s+=.*'.format(key=key)`.
If you require a space before the `=` (that is something that I asked to check, if without space it works or not) you need to put the `+`.
I know that it's almost the same, but it gives more meaning to the pattern (and we don't group anything that we don't use later)

Reply:
With the `\s+` we won't be able to find/update the pattern in HANA config files, if we are using the generic method in shell.py, because the pattern exists there without space 'key=value'

Reply:
I thought that you discarded to have a unique method for both

Reply:
Hi @arbulu89 I have updated the pattern you suggested as am discarding to have the unique method for now